### ISSUE TITLE
validation.undeclaredVariableCheck incorrectly reports jsx <lower.Upper/> tags as undeclared variables

### ISSUE THREAD

**Issue by vadzim** [babel/babel#1754]
```
> babel --optional validation.undeclaredVariableCheck
<a.B/>
/usr/lib/node_modules/babel/node_modules/babel-core/lib/babel/transformation/file/index.js:628
      throw err;
            ^
ReferenceError: Line 1: Reference to undeclared variable "B"
```
Constructions like `<a.b/>` or `<A.b/>` have no problems (of course if A is declared).

**Comment by sebmck** [comment]
`JSXMemberExpression` needs to be checked in `t.isReferenced`.

**Comment by vadzim** [comment]
Ups. `<a.b/>` actually MUST produce `undeclared 'a'` error if `a` is undeclared, because it is compiled into `React.createElement(a.b, ...)`, but it doesn't.

**Comment by sebmck** [comment]
Yep, because there's no special handling for the `JSXMemberExpression` node.

**Comment by sebmck** [comment]
Fixed via 0ed5c5f48062176d42b2ae2c0f64f518b5ca9101.
